src/lib/utils/index.ts

```diff
@@ -19,22 +20,45 @@ const uploadApiEndpoint = `https://zns-backend.netlify.app/.netlify/functions/up
  * @param params Metadata parameters
  * @returns URI to the created Metadata
  */
-
-export const createDomainMetadata = async (params: DomainMetadataParams) => {
-	// upload image to http backend
-
+const getMetadataObject = async (params: DomainMetadataParams) => {
```

Comment:
Consider renaming this helper method to something along the lines of `uploadMetadata` or something similar.

`getMetadataObject` implies that you are fetching or retrieving a stored value, when this method is actually `transforming` (data), and `uploading` (data).